packages/chartjs-plugin/src/plugin.ts

```diff
@@ -0,0 +1,205 @@
+  chart.defaults.plugins.title.font = {
+    weight: "bold",
+    size: 16,
+  };
+
+  chart.defaults.scales.radialLinear.ticks.backdropColor = () => {
+    return getCSSVariableValue("--onyx-color-base-background-tinted");
+  };
+
+  Object.entries(chart.defaults.scales).forEach(([key, scale]) => {
+    // exclude radialLinear scale because it does not support a title
+    if (key === "radialLinear") return;
+    const typedScale = scale as ScaleOptionsByType<
+      Exclude<keyof ScaleTypeRegistry, "radialLinear">
+    >;
+
+    typedScale.title = {
+      ...typedScale.title,
+      color: createColorGetter("--onyx-color-text-icons-neutral-soft"),
+      font: {
+        size: 16,
+      },
+    };
+  });
+};
+
+const plugin: Plugin<ChartType, undefined> = {
+  id: "onyx",
+  /**
+   * We use the "beforeUpdate" hook to apply styles to the individual datasets
+   * of the chart.
+   */
+  beforeLayout: (chart) => {
+    /** Color offset / index so every dataset gets a unique color. */
+    let colorOffset = 0;
+
+    chart.config.data.datasets.forEach((dataset, index) => {
+      const controller = chart.getDatasetMeta(index).controller;
+
+      if (controller instanceof DoughnutController || controller instanceof PolarAreaController) {
+        colorizeDataset(dataset, colorOffset, true);
+        colorOffset += dataset.data.length;
+      } else if (controller) {
+        colorizeDataset(dataset, colorOffset);
+        colorOffset++;
+      }
+    });
+
+    // due to some reason the doughnut and pie charts do not use the default font size
+    // for the scale titles so we need to manually set them here
+    const controller = chart.getDatasetMeta(0).controller;
+    if (controller instanceof DoughnutController || controller instanceof PieController) {
+      Object.values(chart.config.options?.scales ?? {}).forEach((scale) => {
+        if (!scale) return;
+        const typedScale = scale as ScaleOptionsByType<
+          Exclude<keyof ScaleTypeRegistry, "radialLinear">
+        >;
+
+        typedScale.title.font = {
+          ...typedScale.title.font,
+          size: 16,
+        };
+      });
+    }
+  },
+  /**
+   * Update chart whenever the light/dark mode changes so that the colors are updated
+   */
+  afterInit: (chart) => {
+    const darkModeObserver = new MutationObserver((mutations) => {
+      const oldClasses = mutations[0].oldValue ?? "";
+      const newClasses = (mutations[0].target as HTMLElement).className;
+
+      const oldTheme = oldClasses.includes("dark") ? "dark" : "light";
+      const newTheme = newClasses.includes("dark") ? "dark" : "light";
+
+      // update chart so the colors are updated to light/dark mode
+      if (oldTheme !== newTheme) chart.update();
+    });
+
+    const observerOptions: MutationObserverInit = {
+      attributeFilter: ["class"],
+      attributeOldValue: true,
+    };
+
+    darkModeObserver.observe(document.body, observerOptions);
```

Comment:
Add TODO with reference to #618 (use https://css-tricks.com/working-with-javascript-media-queries/#aa-listen-for-changes)